x-pack/test/apm_api_integration/tests/correlations/failed_transactions.ts

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../../common/ftr_provider_context';
+import { registry } from '../../common/registry';
+import { PartialSearchRequest } from '../../../../plugins/apm/server/lib/search_strategies/correlations/search_strategy';
+import { parseBfetchResponse } from '../../common/utils/parse_b_fetch';
+
+export default function ApiTest({ getService }: FtrProviderContext) {
+  const retry = getService('retry');
+  const supertest = getService('supertest');
+
+  const getRequestBody = () => {
+    const partialSearchRequest: PartialSearchRequest = {
+      params: {
+        environment: 'ENVIRONMENT_ALL',
+        start: '2020',
+        end: '2021',
+        kuery: '',
+      },
+    };
+
+    return {
+      batch: [
+        {
+          request: partialSearchRequest,
+          options: { strategy: 'apmFailedTransactionsCorrelationsSearchStrategy' },
+        },
+      ],
+    };
+  };
+
+  registry.when('on trial license without data', { config: 'trial', archives: [] }, () => {
+    it('queries the search strategy and returns results', async () => {
+      const intialResponse = await supertest
+        .post(`/internal/bsearch`)
+        .set('kbn-xsrf', 'foo')
+        .send(getRequestBody());
+
+      expect(intialResponse.status).to.eql(
+        200,
+        `Expected status to be '200', got '${intialResponse.status}'`
+      );
+      expect(intialResponse.body).to.eql(
+        {},
+        `Expected response body to be an empty object, actual response is in the text attribute. Got: '${JSON.stringify(
+          intialResponse.body
+        )}'`
+      );
+
+      const body = parseBfetchResponse(intialResponse)[0];
+
+      expect(typeof body.result).to.be('object');
+      const { result } = body;
+
+      expect(typeof result?.id).to.be('string');
+
+      // pass on id for follow up queries
+      const searchStrategyId = result.id;
+
+      // follow up request body including search strategy ID
+      const reqBody = getRequestBody();
+      reqBody.batch[0].request.id = searchStrategyId;
+
+      let followUpResponse: Record<string, any> = {};
+
+      // continues querying until the search strategy finishes
+      await retry.waitForWithTimeout(
+        'search strategy eventually completes and returns full results',
+        5000,
+        async () => {
+          const response = await supertest
+            .post(`/internal/bsearch`)
+            .set('kbn-xsrf', 'foo')
+            .send(reqBody);
+
+          followUpResponse = parseBfetchResponse(response)[0];
```

Comment:
It looks like you always want to parse the responses made to the `/bsearch` endpoint. In that case, what about creating a bfetchClient which takes the request body and returns the (parsed) response?

```suggestion
          followUpResponse = await bfetchClient(supertest, reqBody);
```

You could also make bFetchClient a service similar to `supertestAsApmWriteUser` and then you can call it without specifying `supertest`:

```suggestion
		  const supertest = context.getService('bfetchClient');
          followUpResponse = await bfetchClient(reqBody);
```

Reply:
Moved this to an item on the 7.16 meta issue: #109220